File: experiments/runs/2026-08-10_15-25-37_claim-office-example-mapping_v3-basic-tdd_opus-5-no-thinking-3/src/priceList.ts
import { ClaimOfficeError } from './types.js';

export interface PriceListEntry {
  insuranceValue: number;
  basePremium: number;
  /** Components are priced in blocks of 3; main items are not. */
  isComponent: boolean;
}

/** The MHPCO price list: insurance value and base premium per main item. */
const MAIN_ITEMS: Record<string, { insuranceValue: number; basePremium: number }> = {
  sword: { insuranceValue: 1000, basePremium: 100 },
  amulet: { insuranceValue: 600, basePremium: 60 },
  staff: { insuranceValue: 800, basePremium: 80 },
  potion: { insuranceValue: 400, basePremium: 40 },
};

const COMPONENTS = ['rune', 'moonstone'];

export const COMPONENT_INSURANCE_VALUE = 250;
export const COMPONENT_BASE_PREMIUM = 25;
/** A block of exactly 3 alike components is offered at a special rate. */
export const BLOCK_SIZE = 3;
export const BLOCK_BASE_PREMIUM = 60;

export function lookUp(type: string): PriceListEntry {
  const mainItem = MAIN_ITEMS[type];
  if (mainItem) {
    return { ...mainItem, isComponent: false };
  }
  if (COMPONENTS.includes(type)) {
    return {
      insuranceValue: COMPONENT_INSURANCE_VALUE,
      basePremium: COMPONENT_BASE_PREMIUM,
      isComponent: true,
    };
  }
  throw new ClaimOfficeError(`unknown item type: ${type}`);
}

File: experiments/runs/2026-09-22_01-57-46_claim-office-python-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/pricelist.py
"""The MHPCO price list of insurable items."""


class UnknownItemType(Exception):
    """Raised when an item type is not on the MHPCO price list."""


COMPONENT_INSURANCE_VALUE = 250
COMPONENT_BASE_PREMIUM = 25

_MAIN_ITEMS = {
    "sword": (1000, 100),
    "amulet": (600, 60),
    "staff": (800, 80),
    "potion": (400, 40),
}
_COMPONENTS = ("rune", "moonstone")


def is_component(item_type):
    return item_type in _COMPONENTS


def insurance_value(item_type):
    return _entry(item_type)[0]


def base_premium(item_type):
    return _entry(item_type)[1]


def _entry(item_type):
    if is_component(item_type):
        return COMPONENT_INSURANCE_VALUE, COMPONENT_BASE_PREMIUM
    if item_type not in _MAIN_ITEMS:
        raise UnknownItemType(f"unknown item type: {item_type}")
    return _MAIN_ITEMS[item_type]
